pkg/api/vlabs/types.go

```diff
@@ -486,7 +486,7 @@ type AgentPoolProfile struct {
 	Name                                string               `json:"name" validate:"required"`
 	Count                               int                  `json:"count" validate:"required,min=1,max=100"`
 	VMSize                              string               `json:"vmSize" validate:"required"`
-	OSDiskSizeGB                        int                  `json:"osDiskSizeGB,omitempty" validate:"min=0,max=1023"`
+	OSDiskSizeGB                        int                  `json:"osDiskSizeGB,omitempty" validate:"min=0,max=2048"`
```

Comment:
Should we update the `DiskSizesGB` property spec as well?

Reply:
from this document:
looks like it's 32T now.

will change it.

Reply:
>  Each data disk has a maximum capacity of 32,767 gibibytes (GiB).